react/components/Billing/components/BillingForm/index.js

```diff
@@ -39,20 +39,32 @@ class BillingForm extends PureComponent {
   static propTypes = {
     subscribeToPremium: PropTypes.func.isRequired,
     cancelPremiumSubscription: PropTypes.func.isRequired,
-
     applyCouponToSubscription: PropTypes.func.isRequired,
+    onSuccess: PropTypes.func,
     me: propType(billingFormFragment).isRequired,
+    plan_id: PropTypes.string,
+  }
+
+  static defaultProps = {
+    plan_id: null,
+    onSuccess: () => null,
   }
 
   state = {
     mode: 'resting',
     errorMessage: null,
     operations: [],
 
-    plan_id: this.props.me.customer.plan.id,
+    plan_id: this.props.plan_id || this.props.me.customer.plan.id,
```

Comment:
I don't think this needs to be here if you are calling handlePlan in the `componentDidMount`?

Reply:
And this too